store/postgres/src/detail.rs

```diff
@@ -360,3 +373,95 @@ pub fn deployment_entity(
 
     SubgraphDeploymentEntity::try_from(StoredDeploymentEntity(detail, manifest))
 }
+
+#[derive(Queryable, Identifiable, Insertable)]
+#[table_name = "graph_node_versions"]
+pub struct GraphNodeVersion {
+    pub id: i32,
+    pub git_commit_hash: String,
+    pub git_repository_dirty: bool,
+    pub crate_version: String,
+    pub major: i32,
+    pub minor: i32,
+    pub patch: i32,
+    pub pre_release: String,
+    pub rustc_version: String,
+    pub rustc_host: String,
+    pub rustc_channel: String,
+}
+
+impl GraphNodeVersion {
+    pub(crate) fn create_or_get(conn: &PgConnection) -> anyhow::Result<i32> {
+        let git_repository_path =
+            GraphNodeVersion::find_git_repository_path(std::path::Path::new(CARGO_MANIFEST_DIR))?;
```

Comment:
Err, `lazy_static` won't do here since it is only evaluated at runtime. The information that goes into the current `GraphNodeVersion` really needs to be `'static` - it doesn't have to be as that struct, but all the values need to be `'static`, like the major and minor version etc. already are.

Reply:
I agree. 

I was using the `git2` library, but it only works at runtime.
I'll try to swap it for `git-testament`, which runs at build time. 

One possible drawback is that we may not be able to fetch `rustc` information since the crate I'm using (`rustc_version`) is at the same spot as `git2`, as it still would need to call to `rustc` at runtime, which might not be present in non-development builds.